chapters/api_concepts.txt

```diff
@@ -412,11 +412,81 @@ Index>> appendix.
 === Arrays and Memory Ownership
 :array: <<api_concepts_memory, array>>
 
-`ANARIArray` handles represent data arrays in memory, which are shared with
-device implementations. The memory shared with ANARI can be either owned
-by the application, have ownership transferred to the device via a deleter
-callback, or be managed by the device. Input data has dimensions encoded
-in its handle type, where each dimension can have a sparse stride.
+There are two methods of expressing array data on objects: directly mapping
+array elements on an object parameter in order to write data elements, or
+creating array objects represented by a handle.
+
+The first method lets applications directly map a device-internally allocated
+1, 2, or 3 dimensional array data on any given object using
+
+[source,cpp]
+....
+void *anariMapParameterArray1D(ANARIDevice,
+    ANARIObject,
+    const char *parameterName
+    ANARIDataType elementType,
+    uint64_t numElements);
+
+void *anariMapParameterArray2D(ANARIDevice,
+    ANARIObject,
+    const char *parameterName
+    ANARIDataType elementType,
+    uint64_t numElements1,
+    uint64_t numElements2);
+
+void *anariMapParameterArray3D(ANARIDevice,
+    ANARIObject,
+    const char *parameterName
+    ANARIDataType elementType,
+    uint64_t numElements1,
+    uint64_t numElements2,
+    uint64_t numElements3);
+....
+
+Each of these functions return a densely packed, write-only array for the
+application to fill. Whenever an array is directly mapped, any previously mapped
+array configuration (size, dimensionality, etc.) is discarded in favor of the
+currently mapped array configuration.
+
+Devices are only required to allocate directly mapped buffers for parameters
+corresponding extensions that the device implements. In cases where the
+parameter is unknown by the device, the device is permitted to return `NULL`.
+
+Once the array has been filled, applications signal that the device is free to
+consume the data with
+
+[source,cpp]
+....
+void anariUnmapParameterArray(ANARIDevice, ANARIObject, const char *parameterName);
+....
+
+Directly mapped arrays behave like normal parameters in that the underlying
+array data will not be used in the next frame until the object itself is
+committed with `anariCommitParameters`. Furthermore, using `anariUnsetParameter`
```

Comment:
They are "normal parameters". Since a frame is an object this should probably refer to the operation of "rendering a frame".
> Setting an array parameter by direct mapping and unmapping is equivalent to setting the parameter via `setParameter`. Changes to parameters will only be visible to the next frame rendering operation after `commitParameter` has been called.

Reply:
Addressed in ab76bd4.